include/mxnet/lib_api.h

```diff
@@ -566,16 +609,20 @@ typedef MXReturnValue (*createOpState_t)(std::map<std::string, std::string>,
 class CustomOp {
  public:
   explicit CustomOp(const char* op_name) : name(op_name),
-    forward(NULL), backward(NULL), parse_attrs(NULL), infer_type(NULL),
-    infer_shape(NULL), mutate_inputs(NULL), create_opstate(NULL),
-    isSGop(false) {}
-  ~CustomOp() {}
-  CustomOp& setForward(fcomp_t fcomp) {
-    forward = fcomp;
+    parse_attrs(NULL), infer_type(NULL), infer_shape(NULL), mutate_inputs(NULL),
+    create_opstate(NULL), isSGop(false) {}
+  CustomOp& setForward(fcomp_t fcomp, std::string ctx) {
+    char* cstr = new char[ctx.length()+1];
+    strncpy(cstr, ctx.c_str(), ctx.length()+1);
+    forward_ctx_cstr.push_back(cstr);
```

Comment:
i think we should use const char* here like we do for the op_name

Reply:
changed string in parameter to be const char *, as user gonna pass in a static string, in this way we can avoid a redundant copy